pkg/collector/corechecks/windows_event_log/check.go

```diff
@@ -0,0 +1,294 @@
+	}
+	err = c.validateConfig()
+	if err != nil {
+		return fmt.Errorf("configuration error: %w", err)
+	}
+
+	// Create the event subscription
+	err = c.initSubscription()
+	if err != nil {
+		return fmt.Errorf("failed to initialize event subscription: %w", err)
+	}
+
+	// subscription will be started on first check run.
+
+	return nil
+}
+
+func (c *Check) validateConfig() error {
+	var err error
+	c.eventPriority, err = getEventPriorityFromOption(c.config.instance.EventPriority)
+	if err != nil {
+		return fmt.Errorf("invalid instance config `event_priority`: %w", err)
+	}
+	if isaffirmative(c.config.instance.LegacyMode) && isaffirmative(c.config.instance.LegacyModeV2) {
+		return fmt.Errorf("legacy_mode and legacy_mode_v2 are both true. Each instance must set a single mode to true")
+	}
+	if isaffirmative(c.config.instance.LegacyMode) {
+		// wrap ErrSkipCheckInstance for graceful skipping
+		return fmt.Errorf("%w: unsupported configuration: legacy_mode: true", agentCheck.ErrSkipCheckInstance)
+	}
+	if isaffirmative(c.config.instance.LegacyModeV2) {
+		// wrap ErrSkipCheckInstance for graceful skipping
+		return fmt.Errorf("%w: unsupported configuration: legacy_mode_v2: true", agentCheck.ErrSkipCheckInstance)
+	}
+	if c.config.instance.Timeout.IsSet() {
+		// timeout option is deprecated. Now that the subscription runs in the background in a select
+		// style, a timeout on the "wait for events" operation is no longer applicable.
+		c.Warn("instance config `timeout` is deprecated. It is no longer used by the check and can be removed.")
+	}
+	if val, isSet := c.config.instance.ChannelPath.Get(); !isSet || len(val) == 0 {
+		return fmt.Errorf("instance config `path` must be provided and not be empty")
+	}
+	if val, isSet := c.config.instance.Query.Get(); !isSet || len(val) == 0 {
+		// Query should always be set by this point, but might be ""
+		return fmt.Errorf("instance config `query` if provided must not be empty")
+	}
+	startMode, isSet := c.config.instance.Start.Get()
+	if !isSet || (startMode != "now" && startMode != "oldest") {
+		return fmt.Errorf("invalid instance config `start`: '%s'", startMode)
+	}
+	_, err = evtRPCFlagsFromOption(c.config.instance.AuthType)
+	if err != nil {
+		return fmt.Errorf("invalid instance config `auth_type`: %w", err)
+	}
+
+	if val, isSet := c.config.instance.IncludedMessages.Get(); isSet {
+		c.includedMessages, err = compileRegexPatterns(val)
+		if err != nil {
+			return fmt.Errorf("invalid instance config `included_messages`: %w", err)
+		}
+	}
+
+	if val, isSet := c.config.instance.ExcludedMessages.Get(); isSet {
+		c.excludedMessages, err = compileRegexPatterns(val)
+		if err != nil {
+			return fmt.Errorf("invalid instance config `excluded_messages`: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// Cancel stops the check and releases resources
+func (c *Check) Cancel() {
+	// stop background loop and wait for it to finish
+	c.stopSubscription()
+
+	if c.session != nil {
+		c.session.Close()
+	}
+
```

Comment:
Would it make sense to call `SaveLastBoomark` here as well?

Reply:
there's no need. `fetchEventsLoop` saves the bookmark before it exits.